docs/REVIEWING.md

```diff
@@ -36,6 +36,42 @@ following items should also be checked:
 - [ ] Does it comply with [component spec](specs/component.md)?
 - [ ] Does it comply with the [instrumentation spec](specs/instrumentation.md)?
 
+
+### Checklist - new sink
+
+This checklist is specific for Vector's sink code.
+
+#### Logic
+
+- [ ] Does it work? Do you understand what it is supposed to be doing?
+- [ ] Does the retry logic make sense?
+- [ ] Are the tests testing that the sink is emitting the correct metrics?
+- [ ] Are there integration tests?
+
+#### Code structure
+
+- [ ] Is it using the sink prelude (`use crate::sinks::prelude::*`)?
+- [ ] Is the sink a stream based sink?
+      Check that the return value from `SinkConfig::build` is the return from `VectorSink::from_event_streamsink`.
+- [ ] Is it gated by sensible feature flags?
+- [ ] Is the code modularized into `mod.rs`, `config.rs`, `sink.rs`,  `request_builder.rs`, `service.rs`
+- [ ] Does the code follow our [style guidelines].
+
+#### Documentation
+
+- [ ] Look at the doc preview on Netlify. Does it look good?
+- [ ] Is there a `cue` file linking to `base`?
+- [ ] Is there a markdown file under `/website/content/en/docs/reference/configuration/sinks/`?
```

Comment:
These checks seem like the could be automated too 🤔

Reply:
They could, but I wonder if it would be worth it given how infrequently new sinks get added and we would need to only check that they are correct once?